src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1020,13 +1020,17 @@ public void visitSelect(JCFieldAccess tree) {
                 Symbol sym = tree.sym;
                 switch (sym.getKind()) {
                     case FIELD, ENUM_CONSTANT -> {
-                        FieldRef fr = symbolToFieldRef(sym, qualifierTarget.hasTag(NONE) ?
-                                tree.selected.type : qualifierTarget);
-                        TypeElement resultType = typeToTypeElement(types.memberType(tree.selected.type, sym));
-                        if (sym.isStatic()) {
-                            result = append(CoreOp.fieldLoad(resultType, fr));
+                        if (sym.name.equals(names._this) || sym.name.equals(names._super)) {
```

Comment:
Not sure: don't we need super also for field access? E.g.

```
class A {
   Object o;
}
class B extends A {
    int o; // shadows A.o
    Object superO() { return super.o; }
}
```

Reply:
Although I suppose in my example, we still generate a `getfield` for `super.o`, just with a different symbolic description - so the generated model ends up being fine. I suppose I'm not 100% comfortable with the fact that the model here seems to capture a bytecode distinction (e.g. super calls use a special invocation mode, but super field access do not, so we only model the former), rather than a syntactic difference in the original source (e.g. for both method and field access, the `super` keyword is there).